tests/features/maintenance.feature

```diff
@@ -302,3 +298,76 @@ Feature: Check maintenance mode
     Examples: <lock_type>, <lock_host>
         | lock_type | lock_host  |
         | zookeeper | zookeeper1 |
+
+    Scenario Outline: No splitbrain in maintenance mode after failover
+        Given a "pgconsul" container common config
+        """
+            pgconsul.conf:
+                global:
+                    priority: 0
+                    use_replication_slots: 'yes'
+                    quorum_commit: 'yes'
+                primary:
+                    change_replication_type: 'yes'
+                    primary_switch_checks: 1
+                    sync_replication_in_maintenance: 'no'
+                replica:
+                    allow_potential_data_loss: 'no'
+                    primary_switch_checks: 1
+                    min_failover_timeout: 1
+                    primary_unavailability_timeout: 2
+                    primary_switch_restart: no
+                commands:
+                    generate_recovery_conf: /usr/local/bin/gen_rec_conf_with_slot.sh %m %p
+        """
+        Given a following cluster with "<lock_type>" with replication slots
+        """
+            postgresql1:
+                role: primary
+            postgresql2:
+                role: replica
+                config:
+                    pgconsul.conf:
+                        global:
+                            priority: 2
+            postgresql3:
+                role: replica
+                config:
+                    pgconsul.conf:
+                        global:
+                            priority: 1
+        """
+        Then <lock_type> "<lock_host>" has holder "pgconsul_postgresql1_1.pgconsul_pgconsul_net" for lock "/pgconsul/postgresql/leader"
+        Then container "postgresql2" is in quorum group
+        Then container "postgresql3" is in quorum group
+        Then <lock_type> "<lock_host>" has following values for key "/pgconsul/postgresql/replics_info"
+        """
+          - client_hostname: pgconsul_postgresql2_1.pgconsul_pgconsul_net
+            state: streaming
+            sync_state: quorum
+          - client_hostname: pgconsul_postgresql3_1.pgconsul_pgconsul_net
+            state: streaming
+            sync_state: quorum
+        """
+        When we gracefully stop "pgconsul" in container "postgresql1"
+        When we disconnect from network container "postgresql1"
+        Then one of the containers "postgresql2,postgresql3" became a primary
```

Comment:
NIT: It's very difficult to get without reading steps code, that this step is actually remembering container, which is later referenced as "remembered container".
I suggest adding "and we remember that container" to step text for be more clear.